src/main/java/dci/j24e01/OurShop/controllers/ProductController.java

```diff
@@ -18,14 +20,23 @@ public class ProductController {
 
     @Autowired
     CategoryDAO categoryDAO;
-
+    @Autowired
+    private ProductDAOImpl productDAOImpl;
```

Comment:
instead of autowiring the DAOImpl, you could define the getCategoriesPaginated and getTotalPages in the DAO

Reply:
Ok